core/camel-core/src/test/java/org/apache/camel/component/file/FileProducerCharsetUTFtoISOConvertBodyToTest.java

```diff
@@ -35,12 +39,27 @@ class FileProducerCharsetUTFtoISOConvertBodyToTest extends ContextTestSupport {
 
     private static final String DATA = "ABC\u00e6";
 
-    @Test
-    void testFileProducerCharsetUTFtoISOConvertBodyTo() throws Exception {
+    @BeforeEach
+    void writeTestData() {
         try (OutputStream fos = Files.newOutputStream(testFile("input.txt"))) {
             fos.write(DATA.getBytes(StandardCharsets.UTF_8));
+        } catch (IOException e) {
+            fail("The test cannot run due to: " + e.getMessage());
         }
+    }
 
+
+    @AfterEach
+    void cleanupFile() {
+        try {
+            Files.delete(testFile("output.txt"));
+        } catch (IOException e) {
+            fail("The test cannot run due to an error cleaning up: " + e.getMessage());
+        }
+    }
+
+    @RepeatedTest(10)
```

Comment:
Is it related?

Reply:
Yes it is. This test was flaky on some environment. I want to make sure it runs well after this change.